src/WpBookingsUiModule.php

```diff
@@ -0,0 +1,201 @@
+            $this->_adminInit($eventManager, $templateManager);
+        });
+
+        $eventManager->attach('admin_menu', function () use ($templateManager) {
+            $this->_adminMenu($templateManager);
+        });
+    }
+
+    /**
+     * Register hook on admin init which will register everything else.
+     *
+     * @param EventManagerInterface $eventManager
+     * @param TemplateManager $templateManager
+     */
+    protected function _adminInit($eventManager, $templateManager)
+    {
+        /*
+         * Add metabox with availabilities configuration to
+         * service's edit page.
+         */
+        add_meta_box(
+            $this->_getConfig()['metabox']['id'],
+            $this->_getConfig()['metabox']['title'],
+            function () use ($templateManager) {
+                return $this->_renderMetabox($templateManager);
+            },
+            $this->_getConfig()['metabox']['post_type']
+        );
+
+        /*
+         * Add screen options on bookings management page.
+         */
+        $eventManager->attach('screen_settings', function($settings, $screen = null) use ($templateManager) {
+            if ($this->bookingsPageId !== $screen->base)
+                return $settings;
+
+            return $this->_renderBookingsScreenOptions($templateManager);
+        });
+    }
+
+    /**
+     * Register pages in the admin menu.
+     *
+     * @param TemplateManager $templateManager
+     */
+    protected function _adminMenu(TemplateManager $templateManager)
+    {
+        $this->bookingsPageId = add_menu_page(
+            $this->_getConfig()['menu']['root']['page_title'],
+            $this->_getConfig()['menu']['root']['menu_title'],
+            $this->_getConfig()['menu']['root']['capability'],
+            $this->_getConfig()['menu']['root']['menu_slug'],
+            function () use ($templateManager) {
+                echo $this->_renderMainPage($templateManager);
+            },
+            $this->_getConfig()['menu']['root']['icon'],
+            $this->_getConfig()['menu']['root']['position']
+        );
+
+        add_submenu_page(
+            $this->_getConfig()['menu']['root']['menu_slug'],
+            $this->_getConfig()['menu']['settings']['page_title'],
+            $this->_getConfig()['menu']['settings']['menu_title'],
+            $this->_getConfig()['menu']['settings']['capability'],
+            $this->_getConfig()['menu']['settings']['menu_slug'],
+            function () use ($templateManager) {
+                return $this->_renderSettingsPage($templateManager);
+            }
+        );
+
+        add_submenu_page(
+            $this->_getConfig()['menu']['root']['menu_slug'],
+            $this->_getConfig()['menu']['about']['page_title'],
+            $this->_getConfig()['menu']['about']['menu_title'],
+            $this->_getConfig()['menu']['about']['capability'],
+            $this->_getConfig()['menu']['about']['menu_slug'],

```

Comment:
Kindly optimize this code. `$this->_getConfig()` is being called 16 times, 8 of which it is being accessed at `'root'`, 4 at `'settings'` and 4 at `'about'`.

Consider retrieving `$this->_getConfig()` only once and accessing `'root'`, `'settings'` and `'about'` as required.